polynote-kernel/src/main/scala/polynote/kernel/lang/LanguageKernel.scala

```diff
@@ -2,60 +2,42 @@ package polynote.kernel.lang
 
 import java.io.File
 
-import fs2.concurrent.{Enqueue, Queue}
+import fs2.Stream
 import polynote.kernel._
-import polynote.kernel.util.{Publish, RuntimeSymbolTable, SymbolDecl}
+import polynote.kernel.context.{GlobalInfo, RuntimeContext, SymbolDecl}
 
 /**
   * The LanguageKernel runs code in a given language.
   */
-trait LanguageKernel[F[_]] {
-
-  // LanguageKernel is expected to have a reference to the shared runtime symbol table of a notebook
-  val symbolTable: RuntimeSymbolTable
-
-  final type Decl = SymbolDecl[F, symbolTable.global.type]
-
-  def predefCode: Option[String]
+trait LanguageKernel[F[_], G <: GlobalInfo] {
```

Comment:
Idea: Take `G` out of here and instead make it a type parameter to `runCode` et. al.?